### PR TITLE
Let compiler decide list assemblies for CLI import

### PR DESCRIPTION
**Pull request by kant2002**
I put list of assemblies for resolving CLI import in the `TranslationUnitContext`. I need this for basic runtime library support, since resolution should happens from runtime library, and I do not want to use `Assembly.GetReferencesAssemblies()` purely, because then I cannot run this compiler via NativeAOT.

Other benefit, is that way we explicitly control list of assemblies to resolve CLI import and then this can be controller by developer via command line arguments.

### CODE REVIEW

File: Cesium.CodeGen/Contexts/TranslationUnitContext.cs [hunk 1]
```diff
@@ -1,8 +1,9 @@
+using System.Reflection;
 using Mono.Cecil;
 
 namespace Cesium.CodeGen.Contexts;
 
-public record TranslationUnitContext(AssemblyContext AssemblyContext)
+public record TranslationUnitContext(AssemblyContext AssemblyContext, Assembly[] ImportAssemblies)
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Contexts/TranslationUnitContext.cs:6]: The idea is good, but could you please move that list up to the `AssemblyContext`?

I am already preparing the architecture for the case when there are multiple translation units in a single assembly.

File: Cesium.CodeGen/Extensions/TypeSystemEx.cs [hunk 2]
```diff
@@ -16,13 +17,30 @@ public static class TypeSystemEx
...
-        var type = runtimeAssembly.GetType(typeName);
-        if (type == null) return null;
-        var method = type.GetMethod(methodName);
+        var method = FindType(context.ImportAssemblies, typeName, methodName);
         if (method == null) return null;
 
         return context.Module.ImportReference(method);
     }
+
+    private static MethodInfo? FindType(Assembly[] assemblies, string typeName, string methodName)
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Extensions/TypeSystemEx.cs:26]: _(Minor)_ I suggest removing this (both overloads) into `FindMethod`.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
Overall: the changes are right, this is what's needed to be done.

I have only a couple of minor complaints.

File: Cesium.CodeGen/Extensions/TypeSystemEx.cs (hunk 2 above)
Comment (kant2002) [reply Cesium.CodeGen/Extensions/TypeSystemEx.cs:26]: Not clear what do you mean here.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**